GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -359,6 +360,50 @@ void ManageWorktreeSupport()
             }
         }
 
+        private void FillNextPullActionAsDefaultToolStripMenuItems()
+        {
+            var setDefaultPullActionDropDown = (ToolStripDropDownMenu)setDefaultPullButtonActionToolStripMenuItem.DropDown;
+
+            // Show both Check and Image margins in a menu
+            setDefaultPullActionDropDown.ShowImageMargin = true;
+            setDefaultPullActionDropDown.ShowCheckMargin = true;
+
+            // Prevent submenu from closing while options are changed
```

Comment:
Are we replacing one unorthodox behaviour with another? :)

Reply:
I don't have strong feelings about this... like you I set this option to Fetch all once per repo and forget about it. That said, it's nice to see a checkbox appear next to an item and the icon change. Reassures chat a change was made and makes it more visible about what has actually happened. Should I remove it? :)

Reply:
I guess I'm in the opposite camp as I thought the menu shouldn't stay open after I have made my selection... To me it seemed unorthodox
I won't block it though, if everyone thinks it is a good idea

Reply:
I think it is fine.  Click wrong option and you can fix it.  You get feedback by what checkbox is shown.   Anyone can figure out that the menu will go away when I click somewhere else.